MANUAL.md

```diff
@@ -0,0 +1,159 @@
+
+```rust
+extern crate exonum;
+extern crate exonum_testkit;
+extern crate my_service;
+
+use my_service::{MyService, MyTransaction, MySchema};
+use exonum_testkit::TestKitBuilder;
+
+#[test]
+fn test_my_tx() {
+    // Create simple testkit network.
+    let mut testkit = TestKitBuilder::validator()
+        .with_service(MyService::new())
+        .create();
+    // Create transaction.
+    let tx = MyTransaction::new(...);
+    // Commit it into blockchain.
+    testkit.create_block_with_transactions(txvec![tx.clone()]);
+    // Check the expected result
+    let snapshot = testkit.snapshot();
+    let schema = MySchema::new(&snapshot);
+    assert!(schema.is_my_data_checked());
+}
+```
+
+Make sure that you have full coverage of the business logic in the `execute` method of your transactions.
+But if you just want to check the `verify` logic in transaction, you can do it without testkit in simple way like this:
+
+```rust
+let tx = MyTransaction::new(...);
+assert!(tx.verify());
+```
+
+### Api testing
+
+The following steps may helps you.
+
+* Define the `MyServiceApi` trait for the `TestKitApi` structure that covers the whole api of your service.
+* Implement functions that uses transactions to fill your storage with the test data.
+* Create the tests that check all of your endpoints.
+
+```rust
+
+// Api trait definition.
+
+trait MyServiceApi {
+    fn get_public_data(&self) -> ApiResponsePublicData;
+    fn get_private_data(&self) -> ApiResponsePrivateData;
+    fn post_private_data(&self, data: MyPrivateData) -> ApiResponsePostPrivateData;
+}
+
+impl MyServiceApi for TestKitApi {
+    fn get_public_data(&self) -> ApiResponsePublicData {
+        self.get(ApiKind::Service("my_service"), "/v1/first_endpoint")
+    }
+
+    fn get_private_data(&self) -> ApiResponsePublicData {
+        self.get_private(ApiKind::Service("my_service"), "/v1/second_endpoint")
+    }
+
+    fn post_private_data(&self, data: &MyPrivateData) -> ApiResponsePostPrivateData {
+        self.post(
+            ApiKind::Service("my_service"),
+            "v1/third_endpoint",
+            &data,
+        )
+    }
+}
+
+// Api test skeleton
+
+#[test]
+fn my_api_test() {
+    let mut testkit = TestKitBuilder::validator()
+        .with_service(MyService::new())
+        .create();
+    fill_storage_with_data(&mut testkit);
+    // Check api responses
+    let api = testkit.api();
+    assert_eq!(api.get_public_data(), ApiResponsePublicData::new(...));
+    ...
+}
+```
+
+## Advanced usage
+
+Here are examples of more complex and less common cases.
+
+### Oracles testing
+
+The oracle in this case is service which can produce transaction with external data after the commit of block,
+[`exonum-time`][exonum-time] and [`exonum-btc-anchoring`][exonum-btc-anchoring] are examples of these kind of oracles.
+In this way, transactions, which were created during the `handle_commit` execution, will store in `TestKit` memory pool.
```

Comment:
...will **be** store**d**...

Additionally, I'm not sure about the second and the third commas in this sentence. I prefer to remove them. :smile: